### PR TITLE
feat: turn signal as action

### PR DESCRIPTION
**Pull request by koritsky**
- move `turn_signal` from state to action
- generalized logit_processing during policy's `predict()`
- rename `PredictionResultKey.PREDICTION` -> `PredictionResultKey.PREDICTION_VALUE` for clarity
- `logit_bias` params are now **not** saved to `state_dict`

### CODE REVIEW

File: src/rmind/scripts/train.py [hunk 1]
```diff
@@ -3,12 +3,13 @@
 
 import hydra
 import pytorch_lightning as pl
-import wandb
 from hydra.utils import instantiate
 from omegaconf import DictConfig, OmegaConf
 from pytorch_lightning.utilities import rank_zero_only
 from structlog import get_logger
 
+import wandb
```
Comment (egorchakov) [inline src/rmind/scripts/train.py:11]: this got moved b/c it considers `wandb` a local package b/c no longer gitignored?

File: src/rmind/callbacks/logit_bias.py [hunk 2]
```diff
@@ -73,6 +74,21 @@ def on_fit_start(self, trainer: pl.Trainer, pl_module: pl.LightningModule) -> No
...
+                return head.out_features
+            case torchvision.ops.MLP():
+                for layer in reversed(head):
+                    if isinstance(layer, torch.nn.Linear):
+                        return layer.out_features
+                msg = "No Linear layer found in MLP"
+                raise ValueError(msg)
+            case _:
+                msg = f"Unsupported head type: {type(head)}"
+                raise NotImplementedError(msg)
```
Comment (egorchakov) [inline src/rmind/callbacks/logit_bias.py:94]:
```suggestion
            case torch.nn.Sequential():
                try:
                    last_linear_layer = next(
                        layer
                        for layer in reversed(head)
                        if isinstance(layer, torch.nn.Linear)
                    )
                except StopIteration:
                    raise ValueError("...")
                else:
                    return last_linear_layer.out_features
```

File: src/rmind/components/objectives/policy.py [hunk 3]
```diff
@@ -152,37 +151,48 @@ def predict(
             )
 
             logits = self.heads.forward(features, batch_size=[b, 1])
+            if set(logits.keys()) != {Modality.DISCRETE, Modality.CONTINUOUS}:  # pyright: ignore[reportArgumentType]
+                msg = f"Only {Modality.DISCRETE} and {Modality.CONTINUOUS} modalities are supported, got {logits.keys()}"
+                raise ValueError(msg)
```
Comment (egorchakov) [inline src/rmind/components/objectives/policy.py:156]: perhaps define inline funcs with match/case that are then passed to `named_apply` vs enforcing here?

File: src/rmind/models/control_transformer.py [hunk 4]
```diff
@@ -124,6 +124,9 @@ def load_from_checkpoint(  # pyright: ignore[reportIncompatibleMethodOverride]
                         case _:
                             raise NotImplementedError
                 checkpoint[cls.CHECKPOINT_HYPER_PARAMS_KEY] = hparams
+                for key in list(checkpoint["state_dict"].keys()):
+                    if "_logit_bias" in key:
+                        checkpoint["state_dict"].pop(key)
```
Comment (egorchakov) [inline src/rmind/models/control_transformer.py:129]: this essentially implies we always recompute logit bias from the current dataset?

checkpoint surgery feels footgunish, maybe add a flag to `LogitBiasSetter` to be explicit about whether we want to overwrite logit bias instead?

File: .gitignore [hunk 5]
```diff
@@ -131,7 +131,7 @@ dmypy.json
 
 # Experiments
 artifacts
-./wandb/
+wandb
```
[the remark this answers is not in the document]
Comment (koritsky) [reply .gitignore:134]: somehow `wandb/` was not ignored even though `.gitignore` is located in the same root dir as `wandb/`

File: src/rmind/models/control_transformer.py (hunk 4 above)
Comment (koritsky) [reply src/rmind/models/control_transformer.py]: https://github.com/yaak-ai/rmind/blob/68057f50d3ddbab08f282712b82877e67295b774/src/rmind/components/loss.py#L40

Fixed here by setting `persistent=False`. We basically remove `logit_bias` from `state_dict`, hence it's not saved in checkpoint. The problem is we need to recompute `logit_bias` if we want to continue training from checkpoint, but such cases are rare.

File: src/rmind/components/objectives/policy.py [hunk 6]
```diff
@@ -151,78 +157,87 @@ def predict(
...
                     nested_keys=True,
                 ).apply(timestep_padder, batch_size=[b, t])  # pyright: ignore[reportAttributeAccessIssue]
+                for result_key in result_keys
+                & {
+                    PredictionResultKey.PREDICTION_VALUE,
+                    PredictionResultKey.PREDICTION_STD,
+                    PredictionResultKey.PREDICTION_PROBS,
+                    PredictionResultKey.SCORE_LOGPROB,
+                    PredictionResultKey.SCORE_L1,
+                }
```
[the remark this answers is not in the document]
Comment (koritsky) [reply src/rmind/components/objectives/policy.py]: a lot of code repetition and also ruff complains about complexity, but agree: a bit more readable
